src/compliment/sources/local_bindings.clj

```diff
@@ -1,30 +1,58 @@
 (ns compliment.sources.local-bindings
   "Completion source for local bindings introduced by defn, let and the like."
   (:require [compliment.sources :refer [defsource]]
-            [compliment.sources.ns-mappings :refer [var-symbol? dash-matches?]]))
-
-(def let-like-forms '#{let if-let when-let if-some when-some loop with-open
-                       dotimes with-local-vars})
-
-(def defn-like-forms '#{defn defn- fn defmacro defmethod})
-
-(def doseq-like-forms '#{doseq for})
-
-(def letfn-like-forms '#{letfn})
+            [compliment.sources.ns-mappings :refer [var-symbol? dash-matches?]]
+            [compliment.utils :as utils]))
+
+;; syntax-quoted variants are important for when we perform macroexpansion as part of our processing.
+(def let-like-forms #{'let `let
```

Comment:
Perhaps, instead of doing this, it makes sense to always drop the namespace from the symbol and only check for the symbol name. If there are collisions, let it be, may be even better.

I suggest you do it in a separate PR and we'll merge it before this.